knife-supermarket.gemspec

```diff
@@ -14,5 +14,5 @@ Gem::Specification.new do |s|
   s.homepage    = 'https://github.com/cwebberOps/knife-supermarket'
   s.license       = 'Apache 2.0'
   s.require_paths = ["lib"]
-  s.add_dependency "chef", ">= 0.10.10"
+  s.add_dependency "chef", ">= 0.10.10", "<= 12.11.18"
```

Comment:
I'm not sure any of this is required, actually. AFAICT Knife's sub command loader will always favour knife subcommands in the chef gem first. I've added a PR to remove `knife-supermarket` from the chef-dk, and I think just pumping out deprecation warnings in the "old chef" case is probably good enough.